packages/dev/core/test/unit/DeviceInput/babylon.deviceInput.test.ts

```diff
@@ -0,0 +1,226 @@
+import { DeviceSource, DeviceSourceManager, DeviceType, PointerInput } from "core/DeviceInput";
+import type { IDeviceInputSystem } from "core/DeviceInput/InputDevices/inputInterfaces";
+import type { InternalDeviceSourceManager, IObservableManager } from "core/DeviceInput/InputDevices/internalDeviceSourceManager";
+import { NullEngine } from "core/Engines/nullEngine";
+import type { IPointerEvent, IUIEvent, IWheelEvent } from "core/Events";
+import type { Nullable } from "core/types";
+
+/**
+ * This class mocks up the basic functionality required from the DeviceInputSystem
+ */
+class MockDeviceInputSystem implements IDeviceInputSystem {
+    constructor(
+        public onDeviceConnected: Nullable<(deviceType: DeviceType, deviceSlot: number) => void>,
+        public onDeviceDisconnected: Nullable<(deviceType: DeviceType, deviceSlot: number) => void>,
+        public onInputChanged: Nullable<(deviceType: DeviceType, deviceSlot: number, eventData: IUIEvent) => void>
+    ) {}
+    pollInput(deviceType: DeviceType, deviceSlot: number, inputIndex: number): number {
+        if (deviceType === DeviceType.Mouse && deviceSlot === 0 && inputIndex === PointerInput.LeftClick) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+    isDeviceAvailable(deviceType: DeviceType): boolean {
+        return deviceType === DeviceType.Mouse || deviceType === DeviceType.Touch;
+    }
+    dispose(): void {
+        // Do nothing
+    }
+}
+
+class MockInternalDeviceSourceManager {
+    public _refCount = 0;
+
+    constructor() {}
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    public readonly registerManager = (manager: IObservableManager): void => {
+        // Do nothing
+    };
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    public readonly unregisterManager = (manager: IObservableManager): void => {
+        // Do nothing
+    };
+
+    public dispose(): void {}
+}

```

Comment:
Is this necessary? Can we not use the real `InternalDeviceSourceManager`?

Reply:
Since I'm going to mock the implementation of the WebDeviceInputSystem, I should be able to use the standard InternalDeviceSourceManager so this will be removed.